### PR TITLE
expand raptor example

### PR DESCRIPTION
**Pull request by andre-merzky**
now terminates on worker failures

### CODE REVIEW

**Comment by andre-merzky** [comment]
ping @AymenFJA : can you please review the PR independent of applying this in Parsl?

File: src/radical/pilot/session.py [hunk 1]
```diff
@@ -175,6 +175,7 @@ def __init__(self, proxy_url: Optional[str ] = None,
         self._tmgrs    = dict()  # map IDs to tmgr instances
         self._cmgr     = None    # only primary sessions have a cmgr
         self._rm       = None    # resource manager (agent_0 sessions)
+        self._hb       = None    # heartbeat monitor
 
         # this session is either living in the client applicatio or lives in the
```
Comment (AymenFJA) [inline src/radical/pilot/session.py:180]: Typo here: `applicatio` --> `application`

**Review by AymenFJA — APPROVED** [review]
@andre-merzky I left one small comment. Other than that I think it looks good. Thanks.

**Comment by andre-merzky** [comment]
Thanks @AymenFJA

**Pull request merged**